.github/workflows/_bwwl_operations.yml

```diff
@@ -0,0 +1,132 @@
+        with:
+          python-version: "3.11"
+
+      - name: Install bwwl binary
+        run: python -m pip install --upgrade bitwarden_workflow_linter
+
+      - name: Create Branch
+        if: ${{ inputs.operation == 'update' }}
+        id: create-branch
+        run: |
+          NAME="update-actions-$(date +'%Y%m%d-%H%M%S')"
+          git switch -c $NAME
+          echo "name=$NAME" >> $GITHUB_OUTPUT
+
+      - name: Create Branch
+        if: ${{ inputs.operation == 'add' }}
+        id: create-branch
+        env:
+          _ACTION: ${{ inputs.action }}
+        run: |
+          NAME= "add-action-$_ACTION"
+          git switch -c $NAME
+          echo "name=$NAME" >> $GITHUB_OUTPUT
+
+      - name: Run bwwl update
+        if: ${{ inputs.operation == 'update' }}
+        run: bwwl actions update -o src/bitwarden_workflow_linter/default_actions.json
+
+      - name: Run bwwl update
+        if: ${{ inputs.operation == 'add' }}
+        run: bwwl actions add -o src/bitwarden_workflow_linter/default_actions.json ${{ inputs.action }}
+
+      - name: Commit changes
+        env:
+          _PR_BRANCH: ${{ steps.create-branch.outputs.name }}
+          _OP: ${{ inputs.operation }}
+        run: |
+          git commit -m "${{ env._OP }} approved actions" -a
+          git push origin $_PR_BRANCH
+
+      - name: Generate GH App token
+        uses: actions/create-github-app-token@c1a285145b9d317df6ced56c09f525b5c2b6f755 # v1.11.1
+        id: app-token
+        with:
+          app-id: ${{ secrets.GH_APP_ID }}
+          private-key: ${{ secrets.GH_APP_KEY }}
+          owner: ${{ github.repository_owner }}
+
+      - name: Create PR
+        id: create-pr
+        env:
+          GH_TOKEN: ${{ steps.app-token.outputs.token }}
+          _PR_BRANCH: ${{ steps.create-branch.outputs.name }}
+          _TITLE: "bwwl Update/Add Approved Actions"
+          _OP: ${{ inputs.operation }}
+        run: |
+          PR_URL=$(gh pr create --title "$_TITLE" \
+            --base "main" \
+            --head "$_PR_BRANCH" \
+            --label "version:patch" \
+            --label "automated pr" \
+            --body "
+              ## Type of change
+              - [ ] Bug fix
+              - [ ] New feature development
+              - [X] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
+              - [ ] Build/deploy pipeline (DevOps)
+              - [ ] Other
+
+              ## Description
+              - This PR ${{ env._OP }}s to the approved actions for the Bitwarden Workflow Linter.")
+          echo "pr_number=${PR_URL##*/}" >> $GITHUB_OUTPUT
+
+      - name: Approve and Merge PR
+        if: ${{ steps.create-pr.outcome == 'success' }}
+        env:
+          GH_TOKEN: ${{ steps.app-token.outputs.token }}
+          _PR_NUMBER: ${{ steps.create-pr.outputs.pr_number }}
+        run: |
+          gh pr review $_PR_NUMBER --approve
+          gh pr merge $_PR_NUMBER --squash --auto --delete-branch
```

Comment:
How can we approve and merge the PR if you're setting a codeowner for that file?

Reply:
Oh right!. I also like the idea of setting the operation based on trigger. I will update